idpi/test/test_hzerocl.py

```diff
@@ -0,0 +1,75 @@
+import os
+import shutil
+import subprocess
+
+import grib_decoder
+import jinja2
+import numpy as np
+import xarray as xr
+from operators.hzerocl import fhzerocl
+
+
+def test_hzerocl():
+    datadir = "/project/s83c/rz+/icon_data_processing_incubator/data/SWISS"
+    datafile = datadir + "/lfff00000000.ch"
+    cdatafile = datadir + "/lfff00000000c.ch"
+
+    ds = {}
+    grib_decoder.load_data(ds, ["T"], datafile, chunk_size=None)
+    grib_decoder.load_data(ds, ["HHL"], cdatafile, chunk_size=None)
+
+    conf_files = {
+        "inputi": datadir + "/lfff<DDHH>0000.ch",
+        "inputc": datadir + "/lfff00000000c.ch",
+        "output": "<HH>_hzerocl.nc",
+    }
+    out_file = "00_hzerocl.nc"
+    prodfiles = ["fieldextra.diagnostic"]
+
+    testdir = os.path.dirname(os.path.realpath(__file__))
+    tmpdir = testdir + "/tmp"
+    cwd = os.getcwd()
+
+    executable = "/project/s83c/fieldextra/tsa/bin/fieldextra_gnu_opt_omp"
+
+    # create the tmp dir
+    shutil.rmtree(tmpdir, ignore_errors=True)
+    os.mkdir(tmpdir)
+
+    templateLoader = jinja2.FileSystemLoader(searchpath=testdir + "/fe_templates")
+    templateEnv = jinja2.Environment(loader=templateLoader)
+    template = templateEnv.get_template("./test_hzerocl.nl")
+    outputText = template.render(file=conf_files, ready_flags=tmpdir)
+
+    with open(tmpdir + "/test_hzerocl.nl", "w") as nl_file:
+        nl_file.write(outputText)
+
+    # remove output and product files
+    for afile in [out_file] + prodfiles:
+        if os.path.exists(cwd + "/" + afile):
+            os.remove(cwd + "/" + afile)
+
+    subprocess.run([executable, tmpdir + "/test_hzerocl.nl "], check=True)
+    hzerocl = fhzerocl(ds["T"], ds["HHL"])
+
+    fs_ds = xr.open_dataset("00_hzerocl.nc")
+    hzerocl_ref = fs_ds["HZEROCL"].rename({"x_1": "x", "y_1": "y"}).squeeze()
+
+    # For this dataset, on point [74,211] last level (center of volume), i.e. 80 (1 base), T is 273.14966.
```

Comment:
This hack can be removed after the fix in the fieldextra namelist. The missing-value-code can be extracted from the NetCDF file that is created by fieldextra.

Reply:
thanks a lot Petra, Indeed removing the in_field="HZEROCL" fixes the hack. On the one hand the values of undef seems to be compatible with numpy NaN (i.e. the IEEE NaN). On the other hand the hack for value 74,211 is not needed anymore. Fieldextra is doing the same as python here. Which means COSMO was doing something else, but nevermind

Reply:
Yes, COSMO sets HZEROCL to -999. if the height of the zero-degree-isotherm falls below the model Earth's surface, whereas fieldextra sets an undef in this case.

Reply:
I maen: fieldextra sets a NaN, as you saw.